### DISCUSSION TITLE
help me with error on nodejs

### DISCUSSION THREAD

**Discussion by gechternachtv** [pocketbase/pocketbase#769] [Q&A] [answered]
Can someone help me understand what is wrong here? Its working on the frontend perfectly, but when i tried on node, this happened, am i getting something wrong?

My node version is 18.10

```js

import PocketBase from 'pocketbase';
export const client = new PocketBase('http://localhost:8090');

client.users.authViaEmail("myadminemailhere", "myadminpassword")

```
```
ClientResponseError 0: Something went wrong while processing your request.
    at new i (file:///home/gechternacht/Desktop/ibhub/node_modules/pocketbase/dist/pocketbase.es.mjs:1:2078)
    at file:///home/gechternacht/Desktop/ibhub/node_modules/pocketbase/dist/pocketbase.es.mjs:1:29861
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  url: '',
  status: 0,
  data: {},
  isAbort: false,
  originalError: TypeError: fetch failed
      at Object.processResponse (node:internal/deps/undici/undici:6406:34)
      at node:internal/deps/undici/undici:6727:42
      at node:internal/process/task_queues:140:7
      at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
      at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    cause: Error: connect ECONNREFUSED ::1:8090
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16) {
      errno: -111,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '::1',
      port: 8090
    }
  }
}
```

**Comment by benallfree** [comment]
Hi @gechternachtv,

The instructions for nodejs are a little hard to find right now. Here they are: https://github.com/pocketbase/js-sdk#nodejs-via-npm

Basically you must install a `fetch()` polyfill (https://github.com/lquixada/cross-fetch) and possibly an `EventSource` polyfill (https://github.com/EventSource/eventsource). 

I hope that helps!

**Reply by ganigeorgiev** [reply]
`fetch` is available natively in node 18. `EventSource` is required only if you use the realtime service.
The above error is because in the latest node versions `localhost` is resolving to the IPv6 address, instead of the IPv4.

**Comment by ganigeorgiev** [comment, marked as answer]
@gechternachtv This is similar to https://github.com/pocketbase/pocketbase/discussions/272 and https://github.com/pocketbase/pocketbase/issues/258.

Try changing the PocketBase `localhost:8090` address to `127.0.0.1:8090`, eg.:

```js
import PocketBase from 'pocketbase';

export const client = new PocketBase('http://127.0.0.1:8090');
```

In short, the reason this could resolve the issue is that Go dns resolver (and most likely in other languages too) by default will first try to resolve to IPv4 then IPv6 addresses, while in Node17+ for some reason this is the opposite - IPv6 first, then IPv4 (this why in the examples now we are using `127.0.0.1` everywhere).

**Comment by gechternachtv** [comment]
Thanks everyone for the responses, using 'http://127.0.0.1:8090' worked!
